src/Symfony/Component/Serializer/Mapping/Loader/XmlFileLoader.php

```diff
@@ -44,13 +44,12 @@ public function loadClassMetadata(ClassMetadataInterface $classMetadata)
             return false;
         }
 
-        $attributesMetadata = $classMetadata->getAttributesMetadata();
-
         if (isset($this->classes[$classMetadata->getName()])) {
             $xml = $this->classes[$classMetadata->getName()];
 
             foreach ($xml->attribute as $attribute) {
                 $attributeName = (string) $attribute['name'];
+                $attributesMetadata = $classMetadata->getAttributesMetadata();
```

Comment:
This must be done outside of the loop for performance.

Reply:
ok I move out this outside the loop, but I needed to refactor a little bit this method to make it work